### PR TITLE
Plumb WriteBufferManager through JNI

### PR DESCRIPTION
**Pull request by jbhati**
Allow rocks java to explicitly create WriteBufferManager by plumbing it to the native code through JNI.

### CODE REVIEW

**Comment by jbhati** [comment]
@sagar0 can you please review this pull request? Thanks in advance! :)

**Comment by sagar0** [comment]
Thanks! Can you please sign the CLA?

**Comment by jbhati** [comment]
@sagar0 Signed the CLA. Thanks

**Comment by jbhati** [comment]
@sagar0 ping :)

File: java/src/test/java/org/rocksdb/DBOptionsTest.java [hunk 1]
```diff
@@ -424,6 +424,26 @@ public void dbWriteBufferSize() {
     }
   }
 
+  @Test
+  public void setWriteBufferManager() throws RocksDBException {
+    try (final Options opt = new Options()) {
+      final Cache cache = new LRUCache(1 * 1024 * 1024);
+      final long bufferSize = 2000l;
+      final WriteBufferManager writeBufferManager = new WriteBufferManager(bufferSize, cache);
```
Comment (sagar0) [inline java/src/test/java/org/rocksdb/DBOptionsTest.java]: Lets move the instantiation of Cache and writeBufferManager (lines 430 - 432) also into the try block. 
Same comment for the code change in OptionsTest as well.

File: java/src/test/java/org/rocksdb/DBOptionsTest.java [hunk 2]
```diff
@@ -424,6 +424,26 @@ public void dbWriteBufferSize() {
...
+      opt.setWriteBufferManager(writeBufferManager);
+    }
+  }
+
+  @Test
+  public void setWriteBufferManagerWithZeroBufferSize() throws RocksDBException {
+    try (final Options opt = new Options()) {
+      final Cache cache = new LRUCache(1 * 1024 * 1024);
+      final long bufferSize = 0l;
+      final WriteBufferManager writeBufferManager = new WriteBufferManager(bufferSize, cache);
```
Comment (sagar0) [inline java/src/test/java/org/rocksdb/DBOptionsTest.java]: Lets move the instantiation of Cache and writeBufferManager also into the try block.

File: java/src/main/java/org/rocksdb/WriteBufferManager.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package org.rocksdb;
+
+import org.rocksdb.Cache;
+
+/**
+ * Java wrapper over native write_buffer_manager class
+ */
+public class WriteBufferManager extends RocksObject {
```
Comment (sagar0) [inline java/src/main/java/org/rocksdb/WriteBufferManager.java]: May be for later: others can contribute the remaining methods in native WriteBufferManager (like memroy_usage, FreeMem, etc).

**Comment by sagar0** [comment]
I handled the travis and appveyor failures for you; those are green now.

**Comment by siying** [comment]
We don't do it a lot but users should be able to add themselves to https://github.com/facebook/rocksdb/blob/master/AUTHORS if they want.

**Comment by siying** [comment]
Feel free to update AUTHORS, but we need to ask lawyers about non-standard headers.

**Comment by sagar0** [comment]
In this particular case the copyright (and the surrounding author name) was a copy-paste error from merge_operator.cc/MergeOperator.java.

File: java/src/main/java/org/rocksdb/WriteBufferManager.java (hunk 3 above)
Comment (jbhati) [reply java/src/main/java/org/rocksdb/WriteBufferManager.java]: Yeah, maybe we will end up implementing it :)

**Comment by jbhati** [comment]
Thanks for the quick review and taking care of the builds. Appreciate it @Sagar0. Made the requested changes. It is reviewable again.

**Comment by jbhati** [comment]
@sagar0 quick question. I applied this patch to our Manhattan database and I am getting the below error. Was wondering if you know what might be wrong here? 

java.lang.UnsatisfiedLinkError: org.rocksdb.DBOptions.setWriteBufferManager(JJ)V

I have Block cache enabled and WriteBufferManager() creation went through. However, when using DBOptions. java -> setWriteBufferManager(writeBufferManager), it throws the above exception. Can you think of anything on top of your head which might have caused this?

**Comment by jbhati** [comment]
Never mind. Figured it out. Thanks. Will send a PR soon for the fix.

**Pull request closed without merging**